File: Entrada de Dados/Numerica/numero.cpp
#include <iostream>
#include <limits>
using namespace std;

int NUMERO;

int main(void) {
  do {
    cout << "Entre um numero: ";
    if (!(cin >> NUMERO)) {
      cout << "Entrada invalida. Por favor, tente novamente." << endl;
      cin.clear();
      cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }
    else {
      cin.ignore(numeric_limits<streamsize>::max(), '\n');
      break;
    }
  } while (true);
  cout << "Voce informou o numero: " << NUMERO << endl;
  return 0;
}

File: Entrada de Dados/Numerica/numero.lua
local NUMERO = nil
repeat
  io.write("Entre um numero: ")
  local entrada = io.read("*l")
  NUMERO = tonumber(entrada)
  if (NUMERO == nil) then
    print("Entrada invalida. Por favor, tente novamente.")
  end
until NUMERO ~= nil
print("Voce informou o numero: " .. NUMERO)
